src/prediction_service_utils.cpp

```diff
@@ -74,24 +72,14 @@ Status getModelInstance(ovms::ModelManager& manager,
         if (modelInstance == nullptr) {
             return StatusCode::MODEL_VERSION_MISSING;
         }
-
-        return modelInstance->waitForLoaded(WAIT_FOR_MODEL_LOADED_TIMEOUT_MS, modelInstanceUnloadGuardPtr);
-    }
-
-    Status status = StatusCode::MODEL_VERSION_NOT_LOADED_ANYMORE;
-    auto retries = DEFAULT_MODEL_GET_RETRIES;
-    while (status == StatusCode::MODEL_VERSION_NOT_LOADED_ANYMORE && retries--) {
+    } else {
         modelInstance = model->getDefaultModelInstance();
         if (modelInstance == nullptr) {
             return StatusCode::MODEL_VERSION_MISSING;
         }
-        status = modelInstance->waitForLoaded(WAIT_FOR_MODEL_LOADED_TIMEOUT_MS, modelInstanceUnloadGuardPtr);
-        if (status == StatusCode::MODEL_VERSION_NOT_LOADED_ANYMORE) {
-            SPDLOG_INFO("The default version from model {} is retired. Retrying.", modelName);
-        }
     }
 
-    return status;
+    return modelInstance->waitForLoaded(WAIT_FOR_MODEL_LOADED_TIMEOUT_MS, modelInstanceUnloadGuardPtr);
```

Comment:
why we need to wait for loading the model to get the modelInstance reference? it would have consequences in many other functions.

Reply:
Because we don't want to work on instance that is unstable (eg. during loading), otherwise it would cause inference request to fail.

Reply:
I was expecting this method to be used also to report model status. My worry was that it would prevent reporting Loading state. I checked that this method is used only in prediction path so that should be ok.